pkg/ddl/index.go

```diff
@@ -1681,7 +1681,8 @@ func getLocalWriterConfig(indexCnt, writerCnt int) *backend.LocalWriterConfig {
 		return writerCfg
 	}
 
-	availMem := memRoot.MaxMemoryQuota() - memRoot.CurrentUsage()
+	// leave some room for objects overhead
+	availMem := memRoot.MaxMemoryQuota() - memRoot.CurrentUsage() - int64(10*size.MB)
```

Comment:
what this 10M mean?

Reply:
Other place will also call `Consume`, I want to leave some room for them

https://github.com/pingcap/tidb/blob/72d22d60fca2b3d0fd490e5b5ccac9bd56165e36/pkg/ddl/ingest/engine.go#L206

But now I think these call with struct size should be negligible, I will delete them in next commit.

Reply:
I see these struct can have many fields, so maybe they will not be negligible in future. I don't want to delete them now.